File: mobile/iosApp/iosApp/Screens/PaywallScreen.swift
import SwiftUI

struct PaywallScreen: View {
    @Environment(\.dismiss) private var dismiss
    @State private var showToast = false

    private let features = [
        "Rate and vote on FAQs",
        "Submit questions",
        "Post comments",
        "Register an account",
        "Priority support",
    ]

    var body: some View {
        VStack(spacing: 0) {
            Spacer().frame(height: 32)

            Text("Unlock MyFAQ Pro")
                .font(.title)
                .fontWeight(.bold)
                .multilineTextAlignment(.center)

            Spacer().frame(height: 32)

            VStack(alignment: .leading, spacing: 12) {
                ForEach(features, id: \.self) { feature in
                    Label(feature, systemImage: "checkmark")
                        .font(.body)
                }
            }
            .padding(.horizontal, 24)

            Spacer()

            Button {
                showToast = true
            } label: {
                Text("Annual Plan")
                    .frame(maxWidth: .infinity)
            }
            .buttonStyle(.borderedProminent)
            .controlSize(.large)
            .padding(.horizontal, 24)

            Button {
                showToast = true
            } label: {
                Text("Lifetime Access")
                    .frame(maxWidth: .infinity)
            }
            .buttonStyle(.borderedProminent)
            .controlSize(.large)
            .padding(.horizontal, 24)
            .padding(.top, 12)

            Button("Restore purchases") {
                showToast = true
            }
            .padding(.top, 16)
            .padding(.bottom, 24)
        }
        .navigationTitle("")
        .navigationBarTitleDisplayMode(.inline)
        .overlay(alignment: .bottom) {
            if showToast {
                Text("Coming soon")
                    .font(.subheadline)
                    .padding(.horizontal, 16)
                    .padding(.vertical, 10)
                    .background(.ultraThinMaterial)
                    .clipShape(Capsule())
                    .padding(.bottom, 80)
                    .transition(.move(edge: .bottom).combined(with: .opacity))
                    .onAppear {
                        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
                            withAnimation { showToast = false }
                        }
                    }
            }
        }
        .animation(.easeInOut, value: showToast)
    }
}

File: mobile/androidApp/src/main/kotlin/app/myfaq/android/screens/PaywallScreen.kt
package app.myfaq.android.screens

import android.widget.Toast
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Check
import androidx.compose.material3.Button
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PaywallScreen(onBack: () -> Unit) {
    val context = LocalContext.current
    val comingSoon = { Toast.makeText(context, "Coming soon", Toast.LENGTH_SHORT).show() }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {},
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back")
                    }
                },
            )
        },
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(24.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Spacer(modifier = Modifier.height(24.dp))

            Text(
                text = "Unlock MyFAQ Pro",
                style = MaterialTheme.typography.headlineMedium,
                textAlign = TextAlign.Center,
            )

            Spacer(modifier = Modifier.height(24.dp))

            val features = listOf(
                "Rate and vote on FAQs",
                "Submit questions",
                "Post comments",
                "Register an account",
                "Priority support",
            )
            features.forEach { feature ->
                ProFeatureRow(feature)
                Spacer(modifier = Modifier.height(8.dp))
            }

            Spacer(modifier = Modifier.weight(1f))

            Button(
                onClick = comingSoon,
                modifier = Modifier.fillMaxWidth(),
            ) {
                Text("Annual Plan")
            }

            Spacer(modifier = Modifier.height(12.dp))

            Button(
                onClick = comingSoon,
                modifier = Modifier.fillMaxWidth(),
            ) {
                Text("Lifetime Access")
            }

            Spacer(modifier = Modifier.height(16.dp))

            TextButton(onClick = comingSoon) {
                Text("Restore purchases")
            }
        }
    }
}

@Composable
private fun ProFeatureRow(text: String) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier.fillMaxWidth(),
    ) {
        Icon(
            Icons.Default.Check,
            contentDescription = null,
            tint = MaterialTheme.colorScheme.primary,
            modifier = Modifier.size(20.dp),
        )
        Spacer(modifier = Modifier.width(12.dp))
        Text(
            text = text,
            style = MaterialTheme.typography.bodyLarge,
        )
    }
}
